src/platform/plugins/shared/workflows_execution_engine/server/templating_engine.ts

```
@@ -7,37 +7,42 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
-import nunjucks from 'nunjucks';
+import { Liquid } from 'liquidjs';
 
 export class WorkflowTemplatingEngine {
-  constructor() {}
+  private readonly engine: Liquid;
 
-  public render(template: string, context: Record<string, any>): string {
-    return this.renderNunjucks(template, context);
-  }
-
-  private renderNunjucks(template: string, context: Record<string, any>): string {
-    const env = nunjucks.configure({
-      autoescape: true,
+  constructor() {
+    this.engine = new Liquid({
+      strictFilters: true,
+      strictVariables: false,
     });
 
-    // We can add custom functions to the Nunjucks environment here.
-    // In theory, this could be same as `keep.` functions
-    env.addGlobal('now', function (format: string = 'iso') {
-      const date = new Date();
-      if (format === 'iso') return date.toISOString();
-      if (format === 'locale') return date.toLocaleString();
-      return date;
+    // register json filter that converts JSON string to object
+    this.engine.registerFilter('json', (value: any) => {
+      if (typeof value !== 'string') {
+        return value;
+      }
+      try {
+        return JSON.parse(value);
+      } catch (error) {
+        return value;
+      }
```

Is it required to define the parameter with `value: any` type? Would it be possible to receive `value: string`?
Also, not sure we should silence these errors and return non-parsed values, we should assume the return is always a parsed object/array, maybe throwing a descriptive error would be a better option.

---

> LiquidJS filters are often designed to fail silently or return a default value rather than throwing an exception that stops rendering. I'll keep this behavior for consistency.

I’m not entirely sure that’s accurate. LiquidJS doesn’t universally avoid throwing.

While it does silent failures for unknown filters or undefined variables by default, it absolutely supports throwing errors when a filter encounters invalid input (and some built-in filters do). The `strictFilters` and `strictVariables` options exist precisely to control this behavior.

My main concern here is that if we silently pass through the original string when JSON parsing fails, we’ll likely cause a hard-to-debug downstream crash. The next step will receive a string instead of an object, and there won’t be any visible log explaining that the _json_ template parsing failed earlier. That kind of silent failure tends to create mystery bugs in workflow runs.

To make this safer and stay consistent with Liquid’s philosophy, how about we provide two versions of the filter:

- `json` → throws a clear, descriptive error if parsing fails.
- `json?` → lenient version that swallows errors and passes through the original value (or returns a default).

That way, template authors can choose what fits their use case:
- Strict and explicit when data correctness matters.
- Tolerant when they prefer a best-effort render.

This gives us both consistency and clarity without always silencing data corruption.